homeassistant/components/sensor/homematicip.py

```diff
@@ -0,0 +1,418 @@
+
+ATTR_ID = 'device_id'
+ATTR_HOME_ID = 'home_id'
+ATTR_HOME = 'home'
+ATTR_LABEL = 'label'
+ATTR_LASTSTATUS_UPDATE = 'last_status_update'
+ATTR_FIRMWARE = 'status_firmware'
+ATTR_ACTUAL_FIRMWARE = 'actual_firmware'
+ATTR_AVAILABLE_FIRMWARE = 'available_firmware'
+ATTR_LOW_BATTERY = 'low_battery'
+ATTR_UNREACHABLE = 'not_reachable'
+ATTR_SABOTAGE = 'sabotage'
+ATTR_UPTODATE = 'up_to_date'
+ATTR_RSSI_DEVICE = 'rssi_device'
+ATTR_RSSI_PEER = 'rssi_peer'
+ATTR_WINDOW = 'window'
+ATTR_ON = 'on'
+ATTR_EVENT_DELAY = 'event_delay'
+ATTR_VALVE_STATE = 'valve_state'
+ATTR_VALVE_POSITION = 'valve_position'
+ATTR_TEMPERATURE_OFFSET = 'temperature_offset'
+
+ATTR_CONNECTED = 'connected'
+ATTR_DUTYCYCLE = 'dutycycle'
+ATTR_CURRENT_APVERSION = 'current_apversion'
+ATTR_AVAILABLE_APVERSION = 'available_apversion'
+ATTR_TIMEZONE_ID = 'timezone_id'
+ATTR_PIN_ASSIGNED = 'pin_assigned'
+ATTR_UPDATE_STATE = 'update_state'
+ATTR_APEXCHANGE_CLIENTID = 'apexchange_clientid'
+ATTR_APEXCHANGE_STATE = 'apexchange_state'
+ATTR_ACCESSPOINT = 'accesspoint'
+
+HMIP_UPTODATE = 'up_to_date'
+HMIP_VALVE_DONE = 'adaption_done'
+HMIP_SABOTAGE = 'sabotage'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the HomematicIP sensors devices."""
+    # pylint: disable=import-error, no-name-in-module
+    from homematicip.device import (HeatingThermostat,
+                                    TemperatureHumiditySensorWithoutDisplay,
+                                    TemperatureHumiditySensorDisplay)
+
+    _LOGGER.info('Setting up HomeMaticIP accespoint & generic devices')
+    homeid = discovery_info['homeid']
+    home = hass.data[DOMAIN][homeid]
+    add_devices([HomematicipAccesspoint(hass, home)])
+    if home.devices is None:
+        return False
+    for device in home.devices:
+        add_devices([HomematicipDeviceStatus(hass, home, device)])
+        if isinstance(device, HeatingThermostat):
+            add_devices([HomematicipHeatingThermostat(hass, home, device)])
+        if isinstance(device, TemperatureHumiditySensorWithoutDisplay):
+            add_devices([HomematicipSensorThermometer(hass, home, device)])
+            add_devices([HomematicipSensorHumidity(hass, home, device)])
+        if isinstance(device, TemperatureHumiditySensorDisplay):
+            add_devices([HomematicipSensorThermometer(hass, home, device)])
+            add_devices([HomematicipSensorHumidity(hass, home, device)])
+    return True
+
+
+class HomematicipGenericDevice(Entity):
+    """Representation of an HomematicIP generic device."""
+
+    def __init__(self, hass, home, device, signal=None):
+        """Initialize the generic device."""
+        self.hass = hass
+        self._home = home
+        self._device = device
+
+        @callback
+        def event_device(event):
+            """Handle device state changes."""
+            _LOGGER.debug('Event Device: %s', self._device.label)
+            self.schedule_update_ha_state(True)
```

Comment:
Don't call a sync method from an async context. This should be `async_schedule_update_ha_state()`